server.js

```diff
@@ -0,0 +1,116 @@
+const express = require("express");
+const mongoose = require("mongoose");
+const cors = require("cors");
+const session = require("express-session");
+
+const User = require("./UserModel");
+const server = express();
+
+mongoose.connect("mongodb://localhost:27017/userAuthdb")
+    .then(() => {
+        console.log('\n*** Connected to database ***\n');
+    })
+    .catch(err =>{
+        console.log(err.message);
+    });
+
+//middleware
+
+const sessionOptions = {
+    secret: "You're killing me, Smalls...",
+    cookie: {
+        maxAge: 60*60*1000 //1 day
+    },
+    httpOnly: true,
+    secure: false,
+    saveUninitialized: true,
+    resave: true,
+    name: "yadayadayada"
+}
+
+function restricted(req, res, next){
+    if(req.session && req.session.username){
+        next();
+    }
+    else{
+        res.status(401).json("You shall not pass!")
+    }
+}
+
+server.use(express.json());
+server.use(cors());
+server.use(session(sessionOptions)); //global middleware
+
+
+server.get('/', (req, res) => {
+    console.log(req.session);
+    if(req.session && req.session.username) {
+        res.status(200).json({message: `welcome back ${req.session.username}`});
+    }
+    else {
+        res.status(401).json({message: "Login to enter"})
+    }});
+
+server.get('/api/users', restricted, (req, res) => {
```

Comment:
I like the use of local middle-ware here...i wonder if the same would work with a router implementation and apply it to all its nested routes...